packages/flame/lib/src/components/spawn_component.dart

```diff
@@ -33,7 +37,16 @@ class SpawnComponent extends Component {
           !(selfPositioning && area != null),
           "Don't set an area when you are using selfPositioning=true",
         ),
+        assert(
+          factory != null || multiFactory != null,
+          'You need to provide a factory or multiFactory to create components',
+        ),
+        assert(
+          !(factory != null && multiFactory != null),
+          'You need to provide either a factory or a multiFactory not both',
+        ),
```

Comment:
Use `xor` instead and have only one assertion.

Reply:
I know, it just felt logical to do another asertion for the other invariant, but you are right, the xor will suffice